internal/services/storage/storage_share_resource.go

```diff
@@ -143,239 +150,424 @@ func resourceStorageShare() *pluginsdk.Resource {
 			},
 		},
 	}
+
+	if !features.FivePointOhBeta() {
+		r.Schema["storage_account_name"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+			ExactlyOneOf: []string{
+				"storage_account_name",
+				"storage_account_id",
+			},
+			Deprecated: "This property has been deprecated and will be replaced by 'storage_account_id' in version 5.0 of the provider.",
+		}
+
+		r.Schema["storage_account_id"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+			ExactlyOneOf: []string{
+				"storage_account_name",
+				"storage_account_id",
+			},
+		}
```

Comment:
Should these also have `Computed` set on them?

Reply:
In the case one is configured, the other will never have a value as they are mutually exclusive. the schema in `azurerm_storage_container` is incorrectly marked as `Computed`, I'll remove that.